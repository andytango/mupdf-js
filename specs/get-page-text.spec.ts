import { forExamplePages, loadExampleFile } from "./helpers/example-file";

describe("getPageText", () => {
  forExamplePages("should get the page text", async (page) => {
    const { mupdf, doc } = await loadExampleFile();
    expect(mupdf.getPageText(doc, page)).toMatchSnapshot();
  });
});
