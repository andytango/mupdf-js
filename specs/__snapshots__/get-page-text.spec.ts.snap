// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getPageText Page [1 / 4] - should get the page text 1`] = `
"Some text with an embedded font!
PNG and JPEG images:

"
`;

exports[`getPageText Page [2 / 4] - should get the page text 1`] = `
"Here is some vector graphics...

And here is some wrapped text...

Lorem ipsum dolor sit amet, consectetur adipiscing elit. Etiam in

suscipit purus. Vestibulum ante ipsum primis in faucibus orci luctus et
ultrices posuere cubilia Curae; Vivamus nec hendrerit felis. Morbi
aliquam facilisis risus eu lacinia. Sed eu leo in turpis fringilla hendrerit.
Ut nec accumsan nisl. Suspendisse rhoncus nisl posuere tortor
tempus et dapibus elit porta. Cras leo neque, elementum a rhoncus ut,
vestibulum non nibh. Phasellus pretium justo turpis. Etiam vulputate,
odio vitae tincidunt ultricies, eros odio dapibus nisi, ut tincidunt lacus
arcu eu elit. Aenean velit erat, vehicula eget lacinia ut, dignissim non
tellus. Aliquam nec lacus mi, sed vestibulum nunc. Suspendisse
potenti. Curabitur vitae sem turpis. Vestibulum sed neque eget dolor
dapibus porttitor at sit amet sem. Fusce a turpis lorem. Vestibulum ante
ipsum primis in faucibus orci luctus et ultrices posuere cubilia Curae;

Mauris at ante tellus. Vestibulum a metus lectus. Praesent tempor

purus a lacus blandit eget gravida ante hendrerit. Cras et eros metus.
Sed commodo malesuada eros, vitae interdum augue semper quis.
Fusce id magna nunc. Curabitur sollicitudin placerat semper. Cras et
mi neque, a dignissim risus. Nulla venenatis porta lacus, vel rhoncus
lectus tempor vitae. Duis sagittis venenatis rutrum. Curabitur tempor
massa tortor.

"
`;

exports[`getPageText Page [3 / 4] - should get the page text 1`] = `
"Rendering some SVG paths...

"
`;

exports[`getPageText Page [4 / 4] - should get the page text 1`] = `
"Here is a link!

One
Two
Three

"
`;
